swift/README.md

```diff
@@ -249,6 +249,44 @@ if let message = message {
 
 **Don't** use variable names like `optionalMessage` or `unwrappedMessage`.
 
+Favour `guard...else` to nested `if...let`
+==============================================
+Swift allows use of the `guard` keyword to leave a scope quickly if a required condition isn't met. Everything you unwrap with a `guard` is then available for the rest of the block. This pattern allows for easier to read code with less nesting.
+
+```swift
+// Bad (pre Swift 2.0 way)
+func myFunc(key: String?) {
+    if let key = key {
+        if let queue = dict[key] {
+            for queueMessage in queue {
+                if let address = queueMessage["address"] as? MessageAddress,
+                    requestJson = queueMessage["requestJson"] as? String {
+                        // do stuff with everything we unwrapped
+                }
+            }
+        }
+    } else {
+        // log an error
+    }
+}
+```
+
+```swift
+// Good (Swift 2.0 way using guards)
+func myFunc(optionalKey: String?) {
+    guard let key = key else {
+        // log an error
+        return
+    }
+    guard let queue = dict[key] else { return }
```

Comment:
Looking through it now I don't actually see us explicitly saying "no" to this. I thought you were right...but I don't see it. For empty closures we allow it to be on the same line but that's the closest thing I can see.